lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -157,16 +169,19 @@ def run(verify_result: true, allow_protected_ns: false, prune: true)
           This means the desired changes were communicated to Kubernetes, but the deploy did not make sure they actually succeeded.
         MSG
         @logger.summary.add_paragraph(ColorizedString.new(warning).yellow)
-        success = true
       end
-    rescue FatalDeploymentError => error
+      ::StatsD.measure('all_resources.duration', StatsD.duration(start), tags: statsd_tags << "status:success")
+      @logger.print_summary(:success)
+    rescue DeploymentTimeoutError => error
       @logger.summary.add_action(error.message)
-      success = false
-    ensure
```

Comment:
By getting rid of the `ensure`, this PR is implementing solution 2 for https://github.com/Shopify/kubernetes-deploy/issues/137.